app/lib/plugin_initializer.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module DiscourseSolved
+  class PluginInitializer
+    attr_reader :plugin
+
+    def initialize(plugin)
+      @plugin = plugin
+    end
+
+    def apply_plugin_api
+      # this method should be overridden by subclasses
+      raise NotImplementedError
+    end
+  end
+
+  class AssignsRemainderForTopicsQuery < PluginInitializer
+    def apply_plugin_api
+      plugin.register_modifier(:assigns_reminder_assigned_topics_query) do |query|
+        next query if !SiteSetting.prevent_assign_notification
```

Comment:
We will need this site setting to be more specific, something like `ignore_solved_topics_in_assigned_reminder`